src/Roastery/Web/RequestLoggingMiddleware.cs

```diff
@@ -41,12 +43,25 @@ public override async Task<HttpResponse> InvokeAsync(HttpRequest request)
         }
     }
 
-    bool LogCompletion(Exception? exception, HttpRequest request, Stopwatch sw, HttpStatusCode statusCode)
+    bool LogCompletion(LoggerActivity? activity, Exception? exception, HttpStatusCode statusCode)
     {
         var level = (int)statusCode >= 500 ? LogEventLevel.Error : LogEventLevel.Information;
-        _logger.Write(level, exception,
-            "HTTP {RequestMethod} {RequestPath} responded {StatusCode} in {Elapsed:0.000} ms",
-            request.Method, request.Path, (int)statusCode, sw.Elapsed.TotalMilliseconds);
+        if (activity?.Activity != null)
```

Comment:
`LoggerActivity` has a top-level `AddProperty()` method, so just `activity.AddProperty("StatusCode", (int)statusCode)` will do this.

`Complete()` accepts an `Exception` as well `level` so this part can be simplified too.